### DISCUSSION TITLE
AX9000 编译报错：WARNING: modpost: suppressed 11 unresolved symbol warnings because there were too many)

### DISCUSSION THREAD

**Discussion by devnakx** [Boos4721/openwrt#138] [Bugs/问题反馈]
终端输出日志如下：
```
/home/devioa/openwrt/staging_dir/host/bin/fakeroot /home/devioa/openwrt/staging_dir/host/bin/bash /home/devioa/openwrt/scripts/ipkg-build -m "" /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-drv-2022-03-07-397c88cf/ipkg-aarch64_cortex-a53/kmod-qca-nss-drv /home/devioa/openwrt/bin/targets/ipq807x/generic/packages
Packaged contents of /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-drv-2022-03-07-397c88cf/ipkg-aarch64_cortex-a53/kmod-qca-nss-drv into /home/devioa/openwrt/bin/targets/ipq807x/generic/packages/kmod-qca-nss-drv_5.15.40+2022-03-07-397c88cf-9_aarch64_cortex-a53.ipk
echo "kmod-qca-nss-drv" >> /home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/pkginfo/qca-nss-drv.default.install
make[3]: Leaving directory '/home/devioa/openwrt/package/qca/nss/qca-nss-drv'
time: package/qca/nss/qca-nss-drv/compile#0.21#0.06#0.24
make[3]: Entering directory '/home/devioa/openwrt/package/qca/nss/qca-nss-clients'
rm -f /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/.built
touch /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/.built_check
make -C "/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.40" pppoe=y bridge-mgr=y vlan-mgr=y CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" M="/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6" EXTRA_CFLAGS="-I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-nss-drv -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-nss-crypto -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-nss-cfi -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-nss-gmac -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-ssdk -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/qca-ssdk/fal -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/nat46" SoC="ipq807x_64" KCFLAGS="-fmacro-prefix-map=/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl=target-aarch64_cortex-a53_musl" HOSTCFLAGS="-O2 -I/home/devioa/openwrt/staging_dir/host/include -I/home/devioa/openwrt/staging_dir/hostpkg/include -I/home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/host/include -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="Boos4721" KBUILD_BUILD_HOST="Boos" KBUILD_BUILD_TIMESTAMP="Wed May 18 05:04:46 2022" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/home/devioa/openwrt/staging_dir/host/lib" KBUILD_HOSTLDLIBS="-L/home/devioa/openwrt/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls= KBUILD_EXTRA_SYMBOLS="/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/cryptodev-linux.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/gpio-button-hotplug.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/qca-nss-drv.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/ath10k-ct.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/mac80211.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/qca-nss-dp.symvers /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/symvers/qca-ssdk.symvers" KERNELRELEASE=5.15.40 modules
make[4]: Entering directory '/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.40'
  CC [M]  /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/bridge/nss_bridge_mgr.o
  LD [M]  /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/bridge/qca-nss-bridge-mgr.o
  CC [M]  /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/nss_vlan_mgr.o
  LD [M]  /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.o
  MODPOST /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers
ERROR: modpost: "nss_unregister_pppoe_session_if" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_get_context" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_msg_init" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_tx_msg_sync" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_register_pppoe_session_if" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_vlan_tx_vsi_detach_msg" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_add_tag_msg" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_set_mac_addr_msg" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_unregister_vlan_if" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_vsi_attach_msg" [/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
WARNING: modpost: suppressed 11 unresolved symbol warnings because there were too many)
make[5]: *** [scripts/Makefile.modpost:134: /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers] Error 1
make[5]: *** Deleting file '/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers'
make[4]: *** [Makefile:1783: modules] Error 2
make[4]: Leaving directory '/home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.40'
make[3]: *** [Makefile:114: /home/devioa/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/.built] Error 2
make[3]: Leaving directory '/home/devioa/openwrt/package/qca/nss/qca-nss-clients'
time: package/qca/nss/qca-nss-clients/compile#1.43#0.17#1.58
    ERROR: package/qca/nss/qca-nss-clients failed to build.
make[2]: *** [package/Makefile:116: package/qca/nss/qca-nss-clients/compile] Error 1
make[2]: Leaving directory '/home/devioa/openwrt'
make[1]: *** [package/Makefile:110: /home/devioa/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/devioa/openwrt'
make: *** [/home/devioa/openwrt/include/toplevel.mk:230：world] 错误 2
```
![1](https://user-images.githubusercontent.com/38558065/169019030-66638f30-9967-4844-9cc7-238d79e9c5e2.png)
已经make distclean后尝试重新编译几次了，都是这样的错误，还有全新编译时也会卡`Enables kernel network stack path for Shortcut Forwarding Engine`需要手动N/Y选择才能继续

**Comment by supsupsuperstar** [comment]
+1，本地，线上都跑了一次，也是这个错。。。
`make[3]: Leaving directory '/workdir/openwrt/package/network/utils/iproute2'
time: package/network/utils/iproute2/tctiny/compile#0.23#0.07#0.31
make[3]: Entering directory '/workdir/openwrt/feeds/Boos/shadowsocksr-libev'
  LD [M]  /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.o
  MODPOST /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers
ERROR: modpost: "nss_unregister_pppoe_session_if" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_get_context" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_msg_init" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_pppoe_tx_msg_sync" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_register_pppoe_session_if" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/pppoe/qca-nss-pppoe.ko] undefined!
ERROR: modpost: "nss_vlan_tx_vsi_detach_msg" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_add_tag_msg" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_set_mac_addr_msg" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_unregister_vlan_if" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
ERROR: modpost: "nss_vlan_tx_vsi_attach_msg" [/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/vlan/qca-nss-vlan.ko] undefined!
make[4]: Leaving directory '/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.40'
make[3]: Leaving directory '/workdir/openwrt/package/qca/nss/qca-nss-clients'
time: package/qca/nss/qca-nss-clients/compile#2.48#0.40#2.97
WARNING: modpost: suppressed 11 unresolved symbol warnings because there were too many)
make[5]: *** [scripts/Makefile.modpost:134: /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers] Error 1
make[5]: *** Deleting file '/workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/Module.symvers'
make[4]: *** [Makefile:1783: modules] Error 2
make[3]: *** [Makefile:114: /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-clients-2022-04-23-7e86daa6/.built] Error 2
    ERROR: package/qca/nss/qca-nss-clients failed to build.
make[2]: *** [package/Makefile:116: package/qca/nss/qca-nss-clients/compile] Error 1
make[2]: *** Waiting for unfinished jobs....
mkdir -p /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/stamp
SHELL= flock /workdir/openwrt/tmp/.root-copy.flock -c 'cp -fpR /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/shadowsocksr-libev-2.5.6/.pkgdir/shadowsocksr-libev-ssr-local/. /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/'
touch /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/stamp/.shadowsocksr-libev-ssr-local_installed
echo "shadowsocksr-libev-ssr-local" >> /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/pkginfo/shadowsocksr-libev.default.install
mkdir -p /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/stamp
SHELL= flock /workdir/openwrt/tmp/.root-copy.flock -c 'cp -fpR /workdir/openwrt/build_dir/target-aarch64_cortex-a53_musl/shadowsocksr-libev-2.5.6/.pkgdir/shadowsocksr-libev-ssr-redir/. /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/'
touch /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/root-ipq807x/stamp/.shadowsocksr-libev-ssr-redir_installed
echo "shadowsocksr-libev-ssr-redir" >> /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/pkginfo/shadowsocksr-libev.default.install
make[3]: Leaving directory '/workdir/openwrt/feeds/Boos/shadowsocksr-libev'
time: package/feeds/Boos/shadowsocksr-libev/compile#0.20#0.05#0.25
make[2]: Leaving directory '/workdir/openwrt'
make[1]: *** [package/Makefile:110: /workdir/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/workdir/openwrt'
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 2
Error: Process completed with exit code 2.`

**Reply by Gaojianli** [reply]
同样的报错+1
